File: C/BinaryTree.c
#include<stdio.h>
#include<stdlib.h>

typedef struct Node {
    int data;
    struct Node *left;
    struct Node *right;
} Node;



int main() {
 
    Node *root = (Node *)malloc(sizeof(Node));
    Node *node1 = (Node *)malloc(sizeof(Node));
    Node *node2 = (Node *)malloc(sizeof(Node));
    Node *node3 = (Node *)malloc(sizeof(Node));


    root->data = 1;
    node1->data = 2;
    node2->data = 3;
    node3->data = 4;

    root->left = node1;
    root->right = node2;
    node1->left = NULL;  // No child
    node1->right = NULL; // No child
    node2->left = node3;
    node2->right = NULL; // No child
    node3->left = NULL;  // No child
    node3->right = NULL; // No child



    return 0;
}

File: C/Tree/BinaryTree.py
class TreeNode:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None



def inOrder(node):
    if node is None:
        return
    
    inOrder(node.left)
    print(node.data)
    inOrder(node.right)


def preOrder(node):
    if node is None:
        return
    
    print(node.data)
    preOrder(node.left)
    preOrder(node.right)

def postOrder(node):
    if node is None:
        return
    postOrder(node.left)
    postOrder(node.right)
    print(node.data)